Character.cpp

```diff
@@ -87,31 +91,51 @@ int Character::takeDamage(int damage)
 }
 
 
-#include <assert>
+#include "Utility.h"
 void Character::attackInternal(Character& other)
 {
     if( other.hitPoints <= 0 )
     {
         /*
         When you defeat another Character: 
-            a) your stats are restored to their initial value if they are lower than it.
+            a) your stats are restored to their initial value if they are lower than it. (hp, armor, atk)
             b) your stats are boosted 10%
-            c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
+            c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character. initialX
       */
-        assert(false);
+        //assert(false);
+        if( initialHitPoints != nullptr; hitPoints < *initialHitPoints )
+            hitPoints = ceil(*initialHitPoints * 1.1);
+        else
+            hitPoints = ceil(hitPoints * 1.1);
+
+        if( initialAttackDamage != nullptr; attackDamage < *initialAttackDamage)
+            attackDamage = ceil(*initialAttackDamage * 1.1);
+        else
+            attackDamage = ceil(attackDamage * 1.1);
+
+        if( initialArmorLevel != nullptr; armor < *initialArmorLevel)
+            armor = ceil(*initialArmorLevel * 1.1);
+        else
+            armor = ceil(armor * 1.1);
+
+        *initialHitPoints = hitPoints;
+        *initialArmorLevel = armor;
+        *initialAttackDamage = attackDamage;
```

Comment:
if all 6 of these member variables hold the same type, see if you can write a member function that will D.R.Y. all of this. 
- [ ] find a D.R.Y refactor solution.